### PR TITLE
fix(unlock-js): split the tests functions into their own files

### LINKED ISSUE #8580: unlock-js testing should have a way to test only a single version
**Is your feature request related to a problem? Please describe.**

We have integration tests for unlock js for all Unlock > PublicLock pairs, which is good. Now we need a way to simply test the latest version of a lock because development becomes slow and painful by running the entire suite - which leads to basically having less tests, less reliable. 

**Describe the solution you'd like**
A way to (uint) test functions in a single version of a lock

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This is a refactor of the unlock-js integration tests. It will allow to run tests separately and make development (and possibly testing) much faster.

What this PR does :

- [x] keep all existing tests and features (test each unlock + lock versions against each other)
- [x] split each unlockjs function into its own file
- [x] allow to run a tests for a single function and a specific lock/unlock version

# Issues

Fixes #8580
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/__tests__/integration/lock/approveBeneficiary.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { itIfErc20 } from '../../helpers/integration'
+import { ethers } from 'hardhat'
+
+let web3Service, chainId, walletService, lock, lockAddress, ERC20
+
+export default ({ isERC20, publicLockVersion }) =>
+  () => {
+    // only v8+
+    if (['v4', 'v6', 'v7'].indexOf(publicLockVersion) === -1) {
```
Comment (julien51) [inline packages/unlock-js/src/__tests__/integration/lock/approveBeneficiary.js:9]: Maybe we do a `itIfVersion` helper? wdyt?

**Comment by julien51** [comment]
I think ttests are _not_ running? (when looking at the CI output I am looking at them but can't see them!)
https://github.com/unlock-protocol/unlock/actions/runs/3282169520/jobs/5405208409

File: packages/unlock-js/src/__tests__/integration/lock/approveBeneficiary.js (hunk 1 above)
Comment (clemsos) [reply packages/unlock-js/src/__tests__/integration/lock/approveBeneficiary.js:9]: yes, had the same idea but wanted to get everything running first

**Comment by clemsos** [comment]
> I think ttests are _not_ running? (when looking at the CI output I am looking at them but can't see them!) https://github.com/unlock-protocol/unlock/actions/runs/3282169520/jobs/5405208409

oh~ seems they havent been running on master either https://github.com/unlock-protocol/unlock/actions/runs/3283836119/jobs/5409056857

**Comment by clemsos** [comment]
likely the `generate:locksmith` task for open api is spinning a subthread that prevents the `test` task from completing. Not sure how https://github.com/unlock-protocol/unlock/blob/ff235f603815d4607ab1b75a61404f5c60d87a53/packages/unlock-js/package.json#L23

**Comment by clemsos** [comment]
```
120 skipped, 4331 passed, 4451 total
```

ok all tests running properly

**Comment by clemsos** [comment]
> ```
> 120 skipped, 4331 passed, 4451 total
> ```

So after this refactoring, it seems we got some more tests running...(is because of the way we now check versions using helpers).

```
 120 skipped, 4696 passed, 4816 total
 ````

File: packages/unlock-js/hardhat.config.js [hunk 2]
```diff
@@ -9,38 +9,93 @@ const { runCLI } = require('jest')
...
-  const testFailures = await new Promise((resolve, reject) => {
-    return runCLI(jestConfig, [process.cwd()])
-      .then((result) => resolve(result))
-      .catch((error) => reject(error))
-  })
+subtask(TASK_JEST_RUN_TESTS).setAction(
+  async ({ watch = false, files = [] }) => {
+    if (files.length) {
+      jestConfig.testMatch = files.map((it) =>
+        it.endsWith('.js') || it.endsWith('*') ? `**/${it}**` : `${it}\\.js$`
```
[the remark this answers is not in the document]
Comment (clemsos) [reply packages/unlock-js/hardhat.config.js:21]: I stole it online somewhere :)

**Review by julien51 — APPROVED** [review]

**Pull request merged**